### PR TITLE
Cleanup/clang format

### PR DESCRIPTION
**Pull request by bill-mccormick-dg**
Please review: @schornakj

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,57 @@
 # Overview
 ODVA confomrant Ethernet/IP interface drivers for Tolomatic stepper and servo controllers, where the ROS node is implimented as an Ethernet/IP adapter. Drivers depend on [odva_ethernetp]('https://github.com/ros-drivers/odva_ethernetip'), a ROS-ready library implimenting the Ethernet/IP protocol.
 
+Topics named _\*\_inputs_ are a direct echo of interface produced values, while _\*\_status_ topics are derived from elements of these inputs. Note that, in the case of _servo_status_, _target_postion_ and _position_error_ require TMI Network Assembly Remapping of Register 1 (Commanded Postion) and Register 2 (Actual Postion Error). For stepper, default mapping is used.
+
 ## Implimentation Limitations
```
Comment (schornakj) [inline README.md]: Spelling: "implementation" (shows up elsewhere in comments too).

File: acsi_eip_driver/srv/acsi_enable.srv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+bool enable
+---
+bool success
```
Comment (schornakj) [inline acsi_eip_driver/srv/acsi_enable.srv]: Could use the generic [SetBool service from std_srvs](http://docs.ros.org/melodic/api/std_srvs/html/srv/SetBool.html) for several of the acsi services.

File: acsi_eip_driver/src/servo_node.cpp [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+using std::cout;
+using std::endl;
+using boost::shared_ptr;
+using eip::socket::TCPSocket;
+using eip::socket::UDPSocket;
+
+using namespace acsi_eip_driver;
+
+int main(int argc, char* argv[])
```
Comment (schornakj) [inline acsi_eip_driver/src/servo_node.cpp:46]: How would you feel about implementing your nodes as classes? That's how the nodes in the ROS-I training material are designed, so I think it's a best-practice pattern ([example template](https://github.com/leggedrobotics/ros_best_practices/blob/master/ros_package_template/src/ros_package_template_node.cpp)).

In summary:
- Make the publishers, subscribers, service clients, service servers, etc. class member variables for a new ServoNode class.
- Have the constructor take the node handle as an argument. Do parameter loading and servo TCP/UDP socket setup in the class constructor
- Put the contents of the while loop into a callback that gets triggered by a timer.
- Have a very minimal main function that just creates a new instance of ServoNode and then spins.

File: stepper_eip_driver/doc/3600-4168_13_ethernetip.pdf [hunk 4]
```diff
@@ -0,0 +1 @@
+/home/wmccormick/tolomatic_ws/src/tolomatic/acsi_eip_driver/doc/3600-4168_13_ethernetip.pdf
```
Comment (schornakj) [inline stepper_eip_driver/doc/3600-4168_13_ethernetip.pdf:1]: Should this be updated to point to some package-relative path?

File: stepper_eip_driver/include/output_assembly.h [hunk 5]
```diff
@@ -19,27 +19,19 @@ using eip::serialization::Serializable;
...
 
-typedef enum
-{
-  HOME        = 0,
-  LEFT,BOTTOM = 1,
-  CENTER      = 2,
-  RIGHT,TOP   = 3
-} MOVE_PROFILE;
-
+typedef enum { HOME = 0, LEFT, BOTTOM = 1, CENTER = 2, RIGHT, TOP = 3 } MOVE_PROFILE;
```
Comment (schornakj) [inline stepper_eip_driver/include/output_assembly.h:34]: What do these refer to?

File: stepper_eip_driver/src/stepper_node.cpp [hunk 6]
```diff
@@ -26,14 +25,15 @@ using eip::socket::UDPSocket;
 
 using namespace stepper_eip_driver;
 
-int main(int argc, char *argv[])
+int main(int argc, char* argv[])
```
Comment (schornakj) [inline stepper_eip_driver/src/stepper_node.cpp:28]: Same comment/suggestion as the servo node about turning this node into a class.

File: README.md [hunk 7]
```diff
@@ -74,3 +108,126 @@ bool stop
 ---
 bool success
 ```
+
+## servo_node
+The servo_node uses the full output assembly, where each service (generally) impliments a specific motion type. For service call parameters velocity, postion, and incriment, the parameter sign will control drive direction. See the manual for more detailed info.
+### Parameters
+- host: The IP address of the controller
+- local_ip: Local IP used for implicit messaging
```
Comment (schornakj) [inline README.md:116]: Does local_ip relate to the network settings of the ROS PC? What should it be set to when I'm running the servo?

File: acsi_eip_driver/src/servo_node.cpp [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+  ros::NodeHandle nh("~");
+
+  bool debug;
+  nh.param<bool>("debug", debug, true);
+  ROS_INFO_STREAM("debug is: " << debug);
+  while (debug)
+  {
+    sleep(1);
+    nh.getParam("debug", debug);
+  }
```
Comment (schornakj) [inline acsi_eip_driver/src/servo_node.cpp]: Do you want to include the "wait here until I connect my debugger" capability in the released version of the driver? If so, the parameter should be documented so it's clear how to use it.

File: acsi_eip_driver/src/servo_node.cpp [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+  ros::Time::init();
+
+  ros::Rate throttle(10);
```
Comment (schornakj) [inline acsi_eip_driver/src/servo_node.cpp]: Could the loop rate be parameterized?

File: stepper_eip_driver/src/stepper_node.cpp [hunk 10]
```diff
@@ -26,14 +25,15 @@ using eip::socket::UDPSocket;
...
 
   bool debug;
   ros::param::param<bool>("debug", debug, false);
   ROS_INFO_STREAM("debug is: " << debug);
-  while(debug) {
+  while (debug)
+  {
     sleep(1);
     ros::param::get("debug", debug);
   }
```
Comment (schornakj) [inline stepper_eip_driver/src/stepper_node.cpp:39]: Keep this here too?

File: stepper_eip_driver/src/stepper_node.cpp [hunk 11]
```diff
@@ -82,7 +81,7 @@ int main(int argc, char *argv[])
 
   try
   {
-    //get status
+    // get status
```
Comment (schornakj) [inline stepper_eip_driver/src/stepper_node.cpp]: What's supposed to go here?

**Review by schornakj — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (bill-mccormick-dg) [reply README.md]: Hmm ... I only found it in one place in the README.

File: acsi_eip_driver/srv/acsi_enable.srv (hunk 2 above)
Comment (bill-mccormick-dg) [reply acsi_eip_driver/srv/acsi_enable.srv]: Maybe Trigger would be better for some moveStop, setHome, moveHome?

Only e-stop and enable have the ability to toggle.

File: acsi_eip_driver/src/servo_node.cpp (hunk 3 above)
Comment (bill-mccormick-dg) [reply acsi_eip_driver/src/servo_node.cpp:46]: I'll take a look. A little hand-holding might speed things.

File: stepper_eip_driver/doc/3600-4168_13_ethernetip.pdf (hunk 4 above)
Comment (bill-mccormick-dg) [reply stepper_eip_driver/doc/3600-4168_13_ethernetip.pdf:1]: I don't think so.

File: stepper_eip_driver/include/output_assembly.h (hunk 5 above)
Comment (bill-mccormick-dg) [reply stepper_eip_driver/include/output_assembly.h:34]: Yea, those should be in your higher level implementation of the stepper interface. Anyway, my PR is only for acsi_servo.

File: acsi_eip_driver/src/servo_node.cpp (hunk 3 above)
Comment (schornakj) [reply acsi_eip_driver/src/servo_node.cpp:46]: It can be a later PR since the stepper and servo nodes work fine as-is.

File: README.md (hunk 7 above)
Comment (bill-mccormick-dg) [reply README.md:116]: It will be needed for implicit messaging

File: acsi_eip_driver/src/servo_node.cpp (hunk 8 above)
Comment (bill-mccormick-dg) [reply acsi_eip_driver/src/servo_node.cpp]: Will comment out

File: acsi_eip_driver/src/servo_node.cpp (hunk 9 above)
Comment (bill-mccormick-dg) [reply acsi_eip_driver/src/servo_node.cpp]: Ok.

File: stepper_eip_driver/src/stepper_node.cpp (hunk 10 above)
Comment (bill-mccormick-dg) [reply stepper_eip_driver/src/stepper_node.cpp:39]: Commented out

File: stepper_eip_driver/src/stepper_node.cpp (hunk 11 above)
Comment (bill-mccormick-dg) [reply stepper_eip_driver/src/stepper_node.cpp]: Removed.

**Review by schornakj — APPROVED** [review]

**Pull request merged**